pkg/pillar/types/dns.go

```diff
@@ -682,31 +685,38 @@ func getLocalAddrListImpl(dns DeviceNetworkStatus,
 }
 
 // Check if an interface name is a port owned by nim
-func IsPort(dns DeviceNetworkStatus, ifname string) bool {
+func IsPort(dns DeviceNetworkStatus, ifname string, usbaddr string, usbprod string, pciaddr string) bool {
 	for _, us := range dns.Ports {
-		if us.IfName != ifname {
+		if ifname != "" && us.IfName != ifname {
 			continue
 		}
-		return true
-	}
-	return false
-}
-
-// IsL3Port checks if an interface name belongs to a port with SystemAdapter attached.
-func IsL3Port(dns DeviceNetworkStatus, ifname string) bool {
-	for _, us := range dns.Ports {
-		if us.IfName != ifname {
+		if us.USBAddr != usbaddr {
```

Comment:
I wonder whether we need to treat empty strings as wildcards in some of these comparisons.

Reply:
Turns out that is at least required for PCIAddr in IsPort since the comparison is between the model aka SystemAdapter, which might have an empty PCIAddr and the discovered PCIAddr on the running system, which is not empty.